java/query/src/main/java/sleeper/query/executor/QueryExecutor.java

```diff
@@ -57,27 +61,35 @@ public class QueryExecutor {
     private final ObjectFactory objectFactory;
     private final StateStore stateStore;
     private final Schema schema;
-    private final ExecutorService executorService;
     private final TableProperties tableProperties;
-    private final Configuration configuration;
+    private final LeafPartitionRecordRetriever recordRetriever;
     private List<Partition> leafPartitions;
     private PartitionTree partitionTree;
     private Map<String, List<String>> partitionToFiles;
+    private LocalDateTime cacheExpireTime = LocalDateTime.now();
```

Comment:
The caching behaviour in this class could be unit tested in terms of what will happen when the cache expires.

The important behaviour seems to be that when the cache expires, it will be able to find files and partitions that were added to the state store since the last time the cache was loaded. Until it expires, it will not see any new files that were added to the state store. That means it will be unable to find the data in those files when it runs a Query.